lhotse/recipes/ami.py

```diff
@@ -0,0 +1,256 @@
+
+
+def make_metadata(data_dir: Pathlike, asr_task: bool) -> Dict[str, AmiMetaData]:
+    metadata = {}
+    anotation_lists = parse_ami_annotations(data_dir / 'annotations.gzip')
+    wav_dir = data_dir / 'wav_db'
+    for audio_path in wav_dir.rglob('*.wav'):
+        audio_idx = audio_path.name
+        audio_name = re.sub(r'\..*$', '', audio_idx)
+        if audio_idx not in anotation_lists:
+            logging.warning(f'No annotation found for {audio_idx}')
+            continue
+        audio_info = torchaudio.info(str(audio_path))[0]
+
+        # Split the raw data into segments
+        anotation = anotation_lists[audio_idx]
+        if asr_task:
+            # For ASR task, we split the audio not only by silences, but also by punctuations
+            for seg_idx, seg_info in enumerate(anotation):
+                for subseg_idx, subseg_info in enumerate(seg_info):
+                    duration = subseg_info.end_time-subseg_info.begin_time
+                    if duration > 0:
+                        metadata[f'{audio_idx}-{seg_idx}-{subseg_idx}'] = AmiMetaData(
+                            audio_path=audio_path,
+                            audio_name=audio_name,
+                            audio_info=audio_info,
+                            text=subseg_info.text,
+                            offset_seconds=subseg_info.begin_time,
+                            duration_seconds=duration
+                        )
+        else:
+            # For VAD task, we simply split the audio by silences
+            current_time = 0.0
+            seg_idx = 0
+            for seg_info in anotation:
+                begin_time = seg_info[0].begin_time
+                end_time = seg_info[-1].end_time
+                if current_time < begin_time:
+                    # Insert a unvoiced segment
+                    metadata[f'{audio_idx}-{seg_idx}u'] = AmiMetaData(
+                        audio_path=audio_path,
+                        audio_name=audio_name,
+                        audio_info=audio_info,
+                        text='',
+                        offset_seconds=current_time,
+                        duration_seconds=(begin_time-current_time)
+                    )
+                    seg_idx += 1
+                # Then the current segment
+                metadata[f'{audio_idx}-{seg_idx}v'] = AmiMetaData(
+                    audio_path=audio_path,
+                    audio_name=audio_name,
+                    audio_info=audio_info,
+                    text='[voice]',
+                    offset_seconds=begin_time,
+                    duration_seconds=(end_time - begin_time)
+                )
+                seg_idx += 1
+                current_time = end_time
+
+    return metadata
+
+
+def prepare_ami(
+        data_dir: Pathlike,
+        output_dir: Pathlike,
+        asr_task: Optional[bool] = True,
+        write_yaml: Optional[bool] = False,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
```

Comment:
I think this needs a bit of documentation since it's not clear what the structure of the output is, e.g. what do the string-values keys correspond to?

Reply:
Could you add a docstring that describes what is exactly returned by this function?

Reply:
I'll add the docstring here.